gitpod-io/template-haskell
# Make Haskell template work with a language server

## Description

This pull request changes the default Haskell environment from "stack" and the Ubuntu packages to using "ghcup". This enables the language server to work, which provides IDE-like capabilities.

It also allows users to upgrade to newer versions of the Haskell toolset much easier.

## Related Issue(s)

Fixes #

## How to test

Create a new Gitpod.io project using this template. (Note that the button in README.md still refers to the original.)

Check that hovering the mouse over a function gives you documentation (i.e. the language server is running).

Check that running "stack run" succeeds and prints "someFunc" on the terminal.

Check that "ghcup" is available as a command.



## Release Notes

```release-note
Change the Haskell template to use `ghcup` and make the language server functional.
```

## Documentation
